pkg/api/router.go

```diff
@@ -29,7 +28,7 @@ import (
 	"github.com/timescale/promscale/pkg/telemetry"
 )
 
-func GenerateRouter(apiConf *Config, promqlConf *query.Config, client *pgclient.Client, store *jaegerStore.Store, reload func() error) (*mux.Router, error) {
+func GenerateRouter(apiConf *Config, promqlConf *query.Config, client *pgclient.Client, store *jaegerStore.Store, authWrapper mux.MiddlewareFunc, reload func() error) (*mux.Router, error) {
```

Comment:
This is obviously a smell, having so many input parameters on a function. The function itself is too big as well.

Lets put a TODO on this to refactor it in the future so when somebody goes to add another parameter, they can see it and not do it 😃